### PR TITLE
engine: resources: nspawn: Remove godbus channel buffer

### PR DESCRIPTION
**Pull request by jonathangold**
https://github.com/godbus/dbus/issues/94 is fixed with
https://github.com/godbus/dbus/pull/105, so the
buffered channel is no longer necessary. The call to
RemoveSignal() was also removed, as Close() handles
closing all open channels.

## Tips:

* please read the style guide before submitting your patch:
[docs/style-guide.md](../docs/style-guide.md)

* commit message titles must be in the form:

```topic: Capitalized message with no trailing period```

or:

```topic, topic2: Capitalized message with no trailing period```

* golang code must be formatted according to the standard, please run:

```
make gofmt		# formats the entire project correctly
```

or format a single golang file correctly:

```
gofmt -w yourcode.go
```

* please rebase your patch against current git master:

```
git checkout master
git pull origin master
git checkout your-feature
git rebase master
git push your-remote your-feature
hub pull-request	# or submit with the github web ui
```

* after a patch review, please ping @purpleidea so we know to re-review:

```
# make changes based on reviews...
git add -p		# add new changes
git commit --amend	# combine with existing commit
git push your-remote your-feature -f
# now ping @purpleidea in the github PR since it doesn't notify us automatically
```

### CODE REVIEW

File: engine/resources/nspawn.go [hunk 1]
```diff
@@ -174,7 +172,6 @@ func (obj *NspawnRes) Watch() error {
 
 	defer close(busChan)
 	defer bus.Close()
-	defer bus.RemoveSignal(busChan)
```
Comment (purpleidea) [inline engine/resources/nspawn.go]: Let's make this clean, and keep this, but put it right after the add Signal command. You can add a comment about why it is or isn't absolutely needed if you want, but it's a nice thing to do for symmetry.

File: engine/resources/nspawn.go [hunk 2]
```diff
@@ -174,7 +172,6 @@ func (obj *NspawnRes) Watch() error {
 
 	defer close(busChan)
```
Comment (purpleidea) [inline engine/resources/nspawn.go]: I think this line should go right after the make() for it. Want to move it there in the same patch please?

Comment (jonathangold) [reply engine/resources/nspawn.go]: Is there any reason I shouldn't also move `bus.Close()` to right under `bus, err :=...`? I don't think the order matters in this case.

Comment (purpleidea) [reply engine/resources/nspawn.go]: Oh you definitely should AFAICT :)

I sometimes need to remind myself that present me is often a better programmer than past me, so as a result, I'm always cleaning up after old mistakes or missed mistakes, and should feel free to change anything that's wrong.

**Comment by purpleidea** [comment]
Merged! Thank you :)

**Pull request merged**